resources/wikia/wikia.wikibits.js

```diff
@@ -109,54 +110,145 @@ var importArticle = (function() {
-		if ( !reportMissing ) {
+		if (!reportMissing) {
 			return;
 		}
 
-		if ( !$.isArray( missing ) ) {
-			missing = [ missing ];
+		if (!$.isArray(articles)) {
+			articles = [articles];
 		}
 
 		// Use BannerNotification to show the error to the user
-		if (window.BannerNotification && (missingLength = missing.length)) {
+		if (window.BannerNotification && (missingLength = articles.length)) {
 			var moreLength = missingLength - 1,
-				message = missingText[ missingLength < 2 ? 'single' : 'multiple' ],
-				more = moreText[ moreLength < 2 ? 'single' : 'multiple' ];
+				baseMessageName = baseText[ missingLength < 2 ? 'single' : 'multiple' ],
+				moreMessageName = moreText[ moreLength < 2 ? 'single' : 'multiple'],
+				message;
+
+			message = mw.message(baseMessageName).params([
+				'"' + articles[0] + '"',
+				mw.message(moreMessageName).params([moreLength]).escaped()
+			]).escaped();
+
+			$(function () {
+				new window.BannerNotification(message, 'error').show();
+			});
+		}
+	}
+
+	function importArticleMissing(missing) {
+		showBannerNotification(missing, missingText);
+	}
 
-			message = message
-				.replace( '%1s', '"' + missing[ 0 ] + '"' )
-				.replace( '%2s', more.replace( '%d', moreLength ) );
+	function importNotJsFailed(missing) {
+		showBannerNotification(missing, notJsText);
+	}
 
-			new window.BannerNotification(message, 'error').show();
+	return {
+		importArticleMissing: importArticleMissing,
+		importNotJsFailed: importNotJsFailed
+	};
+}());
+
+/**
+ * Imports script from provided JS page name
+ * Page has to be in MediaWiki namespace and has .js extension
+ *
+ * Article name can point a page on other wikia (e.g. 'external:otherwikiasubdomain:Pagename.js')
+ *
+ * @param {array} articles Names of pages to import without namespace prefix
+ */
+var importWikiaScriptPages = (function () {
+	var namespacePrefix = 'MediaWiki:',
+		externalPrefix = 'external:',
+		externalPrefixLength = externalPrefix.length;
+
+	function importWikiaScriptPages(articles) {
+		var articlesToImport = [],
+			articlesFailed = [];
+
+		for (var i = 0; i < articles.length; i++) {
+			if (!isJsPage(articles[i])) {
+				articlesFailed.push(namespacePrefix + articles[i]);
+				continue;
+			}
+			if (isExternal(articles[i])) {
+				var articlePreparedName = prepareExternalArticleName(articles[i]);
+				if (articlePreparedName.length === 0) {
+					articlesFailed.push(articles[i]);
+					continue;
+				}
+				articlesToImport.push(articlePreparedName);
+				continue;
+			}
+			articlesToImport.push(namespacePrefix + articles[i]);
+		}
+
+		window.importNotifications.importNotJsFailed(articlesFailed);
+
+		window.importArticles({
+			type: 'script',
+			articles: articlesToImport
+		});
+	}
+
+	function isJsPage(scriptName) {
+		return scriptName.substr(scriptName.length - 3) === '.js';
+	}
+
+	function isExternal(scriptName) {
+		return scriptName.substr(0, externalPrefixLength) === externalPrefix;
```

Comment:
I'm not sure about it. In previous version we had ```server``` variable which was much simpler for me to use